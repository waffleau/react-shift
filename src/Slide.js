import React, {PropTypes, PureComponent} from 'react'

import {Transition} from './Transition'

export class Slide extends PureComponent {
  static propTypes = {
    ...Transition.baseProps,
    direction: PropTypes.oneOf([
      'down',
      'downLeft',
      'downRight',
      'left',
      'right',
      'up',
      'upLeft',
      'upRight'
    ]).isRequired
  }

  getDirectionTranslation() {
    switch (this.props.direction) {
      case 'down': return '0,  100%, 0'
      case 'downLeft': return '-100%,  100%, 0'
      case 'downRight': return '100%,  100%, 0'
      case 'left': return '-100%, 0, 0'
      case 'right': return '100%, 0, 0'
      case 'up': return '0,  -100%, 0'
      case 'upLeft': return '-100%,  -100%, 0'
      case 'upRight': return '100%,  -100%, 0'
      default: return '0, 0, 0'
    }
  }

  getFromValue() {
    return {
      transform: 'translate3d(0, 0, 0)'
    }
  }

  getToValue() {
    return {
      transform: `translate3d(${this.getDirectionTranslation()})`
    }
  }

  render() {
    const {children} = this.props
    return (
      <Transition
        {...this.props}
        easing="easeInOut"
        fromValue={this.getFromValue()}
        toValue={this.getToValue()}>
        {children}
      </Transition>
    )
  }
}

export default Slide
