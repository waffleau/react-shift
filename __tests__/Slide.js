import React from 'react'
import renderer from 'react-test-renderer'

import Slide from '../src/Slide'

test('can transition down', () => {
  const tree = renderer.create(
    <Slide
      active
      direction="down">
      {'Children'}
    </Slide>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('can transition down-left', () => {
  const tree = renderer.create(
    <Slide
      active
      direction="downLeft">
      {'Children'}
    </Slide>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('can transition left', () => {
  const tree = renderer.create(
    <Slide
      active
      direction="left">
      {'Children'}
    </Slide>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('can transition up-left', () => {
  const tree = renderer.create(
    <Slide
      active
      direction="upLeft">
      {'Children'}
    </Slide>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('can transition up', () => {
  const tree = renderer.create(
    <Slide
      active={false}
      direction="up">
      {'Children'}
    </Slide>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('can transition up-right', () => {
  const tree = renderer.create(
    <Slide
      active
      direction="upRight">
      {'Children'}
    </Slide>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('can transition right', () => {
  const tree = renderer.create(
    <Slide
      active
      direction="right">
      {'Children'}
    </Slide>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('can transition down-right', () => {
  const tree = renderer.create(
    <Slide
      active
      direction="downRight">
      {'Children'}
    </Slide>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

test('has no translation when inactive', () => {
  const tree = renderer.create(
    <Slide
      active={false}
      direction="down">
      {'Children'}
    </Slide>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})
