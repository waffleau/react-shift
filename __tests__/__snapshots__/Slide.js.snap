exports[`test can transition down 1`] = `
<div
  style={
    Object {
      "transform": "translate3d(0,  100%, 0)",
      "transition": "transform 600ms ease-in-out",
    }
  }>
  Children
</div>
`;

exports[`test can transition down-left 1`] = `
<div
  style={
    Object {
      "transform": "translate3d(-100%,  100%, 0)",
      "transition": "transform 600ms ease-in-out",
    }
  }>
  Children
</div>
`;

exports[`test can transition down-right 1`] = `
<div
  style={
    Object {
      "transform": "translate3d(100%,  100%, 0)",
      "transition": "transform 600ms ease-in-out",
    }
  }>
  Children
</div>
`;

exports[`test can transition left 1`] = `
<div
  style={
    Object {
      "transform": "translate3d(-100%, 0, 0)",
      "transition": "transform 600ms ease-in-out",
    }
  }>
  Children
</div>
`;

exports[`test can transition right 1`] = `
<div
  style={
    Object {
      "transform": "translate3d(100%, 0, 0)",
      "transition": "transform 600ms ease-in-out",
    }
  }>
  Children
</div>
`;

exports[`test can transition up 1`] = `
<div
  style={
    Object {
      "transform": "translate3d(0, 0, 0)",
      "transition": "transform 600ms ease-in-out",
    }
  }>
  Children
</div>
`;

exports[`test can transition up-left 1`] = `
<div
  style={
    Object {
      "transform": "translate3d(-100%,  -100%, 0)",
      "transition": "transform 600ms ease-in-out",
    }
  }>
  Children
</div>
`;

exports[`test can transition up-right 1`] = `
<div
  style={
    Object {
      "transform": "translate3d(100%,  -100%, 0)",
      "transition": "transform 600ms ease-in-out",
    }
  }>
  Children
</div>
`;

exports[`test has no translation when inactive 1`] = `
<div
  style={
    Object {
      "transform": "translate3d(0, 0, 0)",
      "transition": "transform 600ms ease-in-out",
    }
  }>
  Children
</div>
`;
